lib/elixir/src/elixir_clauses.erl

```diff
@@ -52,7 +52,15 @@ warn_zero_length_guard({{'.', _, [erlang, '==']}, Meta,
   Message = io_lib:format("\"length(~ts) == 0\" is discouraged since it has to "
                           "traverse the whole list to check if it is empty or not. "
                           "Prefer to pattern match on an empty list or use "
-                          "\"~ts == []\" as a guard", [ArgString, ArgString]),
+                          "\"~ts == []\" as a guard.", [ArgString, ArgString]),
+  elixir_errors:warn(?line(Meta), ?key(E, file), Message);
+warn_zero_length_guard({{'.', _, [erlang, '>']}, Meta,
+                        [{{'.', _, [erlang, length]}, _, [Arg]}, 0]}, E) ->
+  ArgString = 'Elixir.Macro':to_string(Arg),
+  Message = io_lib:format("\"length(~ts) > 0\" is discouraged since it has to "
+                          "traverse the whole list to check if it is empty or not. "
```

Comment:
I think just mentioning that the list is empty is enough. I find the "or not" part unnecessary, specially for an error msg that aims to be as concise as possible

Reply:
Additionally I would rephrase my previous suggestion since it is not clear what is the "it" in the "it has to traverse...". I wouldngo with this:
> is discouraged since the whole list needs to be traversed in order to check if it's empty

Reply:
I think we should avoid passive voice in general as we just moved the ambiguity elsewhere. Previously it was unclear what was the "it", now it is unclear who is causing the action altogether.